samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -308,38 +312,32 @@ JobModel readJobModelFromMetadataStore(String zkJobModelVersion) {
    */
   @VisibleForTesting
   void loadMetadataResources(JobModel jobModel) {
-    CoordinatorStreamStore coordinatorStreamStore = null;
     try {
-      // Creates the coordinator stream if it does not exists.
-      coordinatorStreamStore = createCoordinatorStreamStore();
-      coordinatorStreamStore.init();
-
       MetadataResourceUtil metadataResourceUtil = createMetadataResourceUtil(jobModel, getClass().getClassLoader());
       metadataResourceUtil.createResources();
 
-      CoordinatorStreamValueSerde jsonSerde = new CoordinatorStreamValueSerde(SetConfig.TYPE);
-      NamespaceAwareCoordinatorStreamStore configStore =
-          new NamespaceAwareCoordinatorStreamStore(coordinatorStreamStore, SetConfig.TYPE);
-      for (Map.Entry<String, String> entry : config.entrySet()) {
-        byte[] serializedValue = jsonSerde.toBytes(entry.getValue());
-        configStore.put(entry.getKey(), serializedValue);
-      }
+      if (coordinatorStreamStore != null) {
+        CoordinatorStreamValueSerde jsonSerde = new CoordinatorStreamValueSerde(SetConfig.TYPE);
```

Comment:
We'd discussed  offline that publishing the heavy-weight configuration bag one-by-one through metadata-store API in leader-startup sequence had significantly slowed down the processor startup. Please create a JIRA ticket for it and add a TODO here.

Reply:
Will do. I'll link the ticket to move this logic in the timer scheduler.